rasgoql/rasgoql/utils/dbt.py

```diff
@@ -134,16 +134,26 @@ def save_schema_file(
     Writes a table def to a dbt schema file
     """
     filepath = os.path.join(output_directory, 'schema.yml')
-    columns_list = []
-    for row in schema:
-        columns_list.append({"name:": row[0]})
-    model_dict = {"name": model_name, "columns": columns_list}
-    if config_args:
-        model_dict.update({"config": config_args})
-    if not os.path.exists(filepath):
-        schema_definition = {"version": 2, "models": [model_dict]}
-    else:
-        schema_definition = [model_dict]
-    with open(filepath, "a") as _file:
+    existing_schema = None
+    if os.path.exists(filepath):
+        with open(filepath, "r") as _file:
+            existing_schema = yaml.safe_load(_file)
+    schema_definition = existing_schema or {"version": 2, "models": []}
+
+    columns_list = [{"name:": row[0]} for row in schema]
+    model_found = False
+    for m in schema_definition["models"]:
+        if m.get("name") == model_name:
+            m["columns"] = columns_list
+            if config_args:
+                m["config"] = config_args
+            model_found = True
+    if not model_found:
```

Comment:
This is fine, but a slightly cleaner way is with a `for else`: 

```suggestion
    for m in schema_definition["models"]:
        if m.get("name") == model_name:
            m["columns"] = columns_list
            if config_args:
                m["config"] = config_args
            break
    else:
```

However this may not work as expected now if people do have models with the same name (as it would only update the first instance of it.)  Do we have (or need to add) checks to ensure name uniqueness?

Reply:
Thankfully models are unique by name in dbt 😌